src/uu/stty/src/stty.rs

```diff
@@ -697,6 +717,65 @@ fn print_in_save_format(termios: &Termios) {
     println!();
 }
 
+/// GNU stty -g compatibility: restore Termios from the colon-separated hexadecimal representation
+/// produced by print_in_save_format.
+fn parse_save_format(s: &str) -> Result<Termios, Box<dyn UError>> {
+    // Expect four flag values + a variable-length sequence of cc_t values (at least one).
+    let parts: Vec<&str> = s.split(':').collect();
+    if parts.len() < 5 {
+        return Err(USimpleError::new(
+            1,
+            translate!("stty-error-invalid-argument", "arg" => s.to_string()),
+        ));
+    }
+
+    // Parse a hex string into tcflag_t (shared helper) to match the underlying bitflag type.
+    fn parse_hex_tcflag(x: &str, original: &str) -> Result<nix::libc::tcflag_t, Box<dyn UError>> {
+        nix::libc::tcflag_t::from_str_radix(x, 16).map_err(|_| {
+            USimpleError::new(
+                1,
+                translate!("stty-error-invalid-argument", "arg" => original.to_string()),
+            )
+        })
+    }
+
+    let iflags_bits = parse_hex_tcflag(parts[0], s)?;
+    let oflags_bits = parse_hex_tcflag(parts[1], s)?;
+    let cflags_bits = parse_hex_tcflag(parts[2], s)?;
+    let lflags_bits = parse_hex_tcflag(parts[3], s)?;
+
+    // Remaining segments are control_chars.
+    let cc_hex = &parts[4..];
+
+    // Obtain the original termios and overwrite specific fields on top of it to preserve
+    // platform-dependent fields.
+    let mut termios = tcgetattr(std::io::stdout().as_fd())
```

Comment:
This is is assuming that we're opening stdout, but no, it should be the device: `tcgetattr(opts.file.as_fd())`.

So, I think that this function should be instead moved at later point after all the args have been parsed, and it should take a `&Termios`, not returning it once it has been opened in the same way it's done currently